python/ray/tune/trial.py

```diff
@@ -340,3 +354,66 @@ def __str__(self):
         if self.experiment_tag:
             identifier += "_" + self.experiment_tag
         return identifier.replace("/", "_")
+
+    def to_serializable(self):
+        """Memento generator for Trial.
+
+        Sets RUNNING trials to PENDING, and flushes the result logger.
+        Note this can only occur if the trial holds a DISK checkpoint.
+        """
+        if not self._checkpoint.storage == Checkpoint.DISK:
+            raise ValueError("Checkpoint cannot be in-memory.")
+        state = {
+            "checkpoint_at_end": self.checkpoint_at_end,
+            "checkpoint_freq": self.checkpoint_freq,
+            "error_file": self.error_file,
+            "experiment_tag": self.experiment_tag,
+            "last_debug": self.last_debug,
+            "last_result": self.last_result,
+            "last_update_time": self.last_update_time,
+            "local_dir": self.local_dir,
+            "location": self.location,
+            "logdir": self.logdir,
+            "max_failures": self.max_failures,
+            "num_failures": self.num_failures,
+            "result_logger": None,
+            "runner": None,
+            "status": Trial.PENDING
+            if self.status == Trial.RUNNING else self.status,
+            "stopping_criterion": self.stopping_criterion,
+            "trainable_name": self.trainable_name,
+            "trial_id": self.trial_id,
+            "upload_dir": self.upload_dir,
+            "verbose": self.verbose
```

Comment:
Why can't we use __dict__ to get this? This list is too long to be maintainable as is.